codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectAggregator.java

```diff
@@ -151,7 +151,11 @@ protected boolean isAggregated(HttpObject msg) throws Exception {
 
     @Override
     protected boolean isContentLengthInvalid(HttpMessage start, int maxContentLength) {
-        return getContentLength(start, -1L) > maxContentLength;
+        try {
+            return getContentLength(start, -1L) > maxContentLength;
+        } catch (final NumberFormatException e) {
+            return false;
```

Comment:
No, it doesn't. Returning `false` prevents the [oversized message handler from incorrectly being invoked in MessageAggregator](https://github.com/netty/netty/blob/7460d90a67bd9724d8fdac839c384bf36cbbce37/codec/src/main/java/io/netty/handler/codec/MessageAggregator.java#L236-L240), instead [allowing the number format exception to escape because the decoder result will be failure](https://github.com/netty/netty/blob/7460d90a67bd9724d8fdac839c384bf36cbbce37/codec/src/main/java/io/netty/handler/codec/MessageAggregator.java#L242).